rapid7_insight_agent/icon_rapid7_insight_agent/util/graphql_api/api_connection.py

```diff
@@ -0,0 +1,234 @@
+from icon_rapid7_insight_agent.util.graphql_api.region_map import region_map
+from insightconnect_plugin_runtime.exceptions import PluginException, ConnectionTestException
```

Comment:
The API-specific code shouldn't raise PluginExceptions/CTE - let the plugin code handle that. This API code should be able to be torn out of the plugin and reused without needing any plugin-specific code. If you need to, you can create an APIException that can be passed back to the plugin code that contains information the plugin code will need to raise a user-viewable exception.